### ISSUE TITLE
hddtemp sensors not updating correctly

### ISSUE THREAD

**Issue by tsumaru720** [librenms/librenms#5904]
Hi Fellas, 

I noticed that last Thursday, graphs for my hddtemps are not displaying correctly in LibreNMS

https://s3-eu-west-1.amazonaws.com/uploads-eu.hipchat.com/23738/290850/cNy6rMcoRcNJD1h/libre.png
https://s3-eu-west-1.amazonaws.com/uploads-eu.hipchat.com/23738/290850/uS0my0ZLfJXWpFH/upload.png

My daily.sh cron runs at 00:15 every day which is about the time this seems to have occured (as per the graph)

As per request from @laf on IRC - Output of (snipped ofc):

`./poller.php -r -f -m applications,unix-agent  -h HOSTNAME -d`

http://pastebin.com/y7sCfhK7

Permissions appear to be correct (and the files are being updated, by rrdcache) however it seems zero value is being inserted into the RRDs themselves - no other issues with RRD files as far as I can tell

```
# ls -alctr | grep hddtemp-4
-rw-rw-r--  1 librenms librenms  125000 Feb 15 11:05 sensor-temperature-hddtemp-4.rrd

# rrdtool fetch sensor-temperature-hddtemp-4.rrd AVERAGE -s 1486598400 -e 1486602000
                         sensor

1486598700: 3.0000000000e+01
1486599000: 3.0000000000e+01
1486599300: 1.2000000000e+00
1486599600: 0.0000000000e+00
1486599900: 0.0000000000e+00
1486600200: 0.0000000000e+00
1486600500: 0.0000000000e+00
1486600800: 0.0000000000e+00
1486601100: 0.0000000000e+00
1486601400: 0.0000000000e+00
1486601700: 0.0000000000e+00
1486602000: 0.0000000000e+00
1486602300: 0.0000000000e+00
```

As evident by the paste, the temps seem to be exposed correctly to LibreNMS

- [x] Is your install up to date? **within last 24 hours**

```
# date
Wed Feb 15 11:01:49 GMT 2017

[WARN]  Your install is out of date, last update: Wed, 15 Feb 2017 00:06:00 +0000
```

- [x] Please include all of the information between the `====================================` section of `./validate.php` which you can run from the cli.

Component | Version
--------- | -------
LibreNMS  | 49b5844346bdfe54616a0e838a7ef634b0e7bdd1
DB Schema | 164
PHP       | 5.5.9-1ubuntu4.21
MySQL     | 5.5.53-0ubuntu0.14.04.1
RRDTool   | 1.4.7
SNMP      | NET-SNMP 5.7.2

- [ ] Please provide ALL info asked for [here](http://docs.librenms.org/Support/FAQ/#faq20).

n/a

- [x] Please provide as much detail as possible.
- [x] Please do NOT post more than 10 lines of debug information here, use a pastebin service or GitHub Gists.

**Comment by laf** [comment]
Can you also provide the output of `./poller.php -h HOSTNAME -d -r -f -m sensors`

**Comment by tsumaru720** [comment]
http://pastebin.com/FSvrHg0x

**Comment by craig** [comment]
I've run into a similar issue and found this before filing a new bug - I'm 95% sure it's the same. Eventlog data from temperature sensors isn't being logged in logs/librenms.log I got:

> 2017-02-16 03:10:04 MySQL Error: Unknown column 'temperature' in 'field list' (INSERT INTO `eventlog` (`host`,`device_id`,`reference`,`type`,`datetime`,`severity`,`message`)  VALUES ('1','1','5',temperature,NOW(),'4',' Core 1 above threshold: 44  (> 41.6 )'))

When running the poller command you send, it (of course) also appears:

> [RRD Disabled]Alerting for TESTHOST Core 2
> SQL[INSERT INTO `eventlog` (`host`,`device_id`,`reference`,`type`,`datetime`,`severity`,`message`)  VALUES ('1','1','6',temperature,NOW(),'4',' Core 2 above threshold: 43  (> 40 )')] 

temperature is not quoted, if you use that sql query with quotes, it works fine.
I've been trying to pinpoint it down and found includes/polling/functions.inc.php, but now it's time for bed. Hopefully this is helpful, good night.

**Comment by laf** [comment]
Please try this patch:

https://gist.githubusercontent.com/laf/708af2480a30f6a5a8a4cd99fc0280c6/raw/ce5176d579536c5881af2b5b1e6f2e0db01fd89f/gistfile1.txt

**Comment by tsumaru720** [comment]
@laf Looks to still be inserting 0's into RRDs.
Ran poller manually 

./poller.php -h HOSTNAME -d

pertinent info: http://pastebin.com/QX9b8UAc

**Comment by exarv** [comment]
Maybe my PR #5930  fixes this?

**Comment by tsumaru720** [comment]
Looking at comments on #5930 which suggests this is fixed so I updated to 1140be1a01d3974c1c15b6661af42ee0437e30b2 and it does seem to be working again

**Comment by geordish** [comment]
18:25 < Demented-Idiot> how do i re-open an issue? #5904 came back :(

Please provide all information asked for here: http://docs.librenms.org/Support/FAQ/#faq20
